packages/blaze-tx/src/TxBuilder.ts

```diff
@@ -1607,14 +1608,25 @@ export class TxBuilder {
         this.body.setCollateral(collateralSet);
       }
 
-      this.body.setCollateralReturn(
```

Comment:
I think it's probably better to:
- split checkAndAlterOutput into two sub functions, one which sets coin to the minUTxO, and one which checks that the UTxO is valid
- modify the coin selection so it will never select a UTxO that doesn't have `minUTxO(value - collateral) + collateral` ADA
- call *only* the `checkOutput` function here

Reply:
Yeah that seems cleaner, I'll update.

Reply:
Updated: https://github.com/butaneprotocol/blaze-cardano/pull/255/commits/f2ad0962be4041994505116d520f00808b35162d